### PR TITLE
Enable deleting users

### PR DESCRIPTION
**Pull request by kellyi**
## Overview

- enable users to be deleted through the Django admin by adding a
missing `allauth.socialaccount` app to installed apps
- update migration to make Contributor a nullable field on FacilityLists
and to set the value to null when the Contributor is deleted
- adjust Facility get_contributors method to skip if
match.facility_list.contributor is None
- add `delete_contributor` Django management command that take an email
address as its argument and deletes the User, Contributor, and
EmailAddress objects associated with the email address

Connects #305 
Connects #269

## Demo

Screen Shot 2019-03-18 at 5 36 31 PM

->

Screen Shot 2019-03-18 at 5 37 31 PM

->

Screen Shot 2019-03-18 at 5 37 12 PM

-> (later, when added from another list)

Screen Shot 2019-03-18 at 5 49 19 PM

## Testing Instructions

- get this branch, then `./scripts/resetdb`
- register for a new user through the UI, then upload list 11 and batch_process all steps to create some facities
- search for some facilities and verify that they appear and that your user/list is displayed among the contributors on its details page
- run `./scripts/manage delete_contributor --email <YOUR_EMAIL>` and verify that it succeeds
- visit the map page and search for facilities. Verify that that works and then visit the details page for a few facilities to observe that the now have no contributors
- register for a new user with a different email address, then upload list 11 again, process it, then look at facilities details pages and verify that the new user is listed among the contributors
- verify that you can delete a user through the Django admin

### CODE REVIEW

**Comment by jwalgran** [comment]
We test users registering accounts on staging and over 1000 new facilities have appeared on the staging map. It would be a little disruptive to reset the database at this time, so I would like to consider creating actual migrations for these changes, and then possibly squashing them back down into single migration later.

File: src/django/api/management/commands/delete_contributor.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from allauth.account.models import EmailAddress
+
+from api.models import User, Contributor
+
+
+class Command(BaseCommand):
```
Comment (jwalgran) [inline src/django/api/management/commands/delete_contributor.py]: I would like to consider giving this the more detailed name `delete_user_and_contributor` as a way of reinforcing that we are deleting multiple things.

**Comment by kellyi** [comment]
> so I would like to consider creating actual migrations for these changes, and then possibly squashing them back down into single migration later.

:+1: Sounds good to me. I'll update this to create a new migration rather than replacing the prior one.

File: src/django/api/management/commands/delete_contributor.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        group.add_argument('-e', '--email',
+                           required=True,
+                           help='The user\'s email address')
+
+    def handle(self, *args, **options):
+        email = options['email']
+
+        try:
+            user = User.objects.get(email=email)
+            contributor = Contributor.objects.get(admin=user)
```
Comment (jwalgran) [inline src/django/api/management/commands/delete_contributor.py]: We should consider moving the some exception handling up to this contributor fetch. It is possible to have a user account without a related contributor. If that is the case, we still want to delete the user account.

**Comment by kellyi** [comment]
dab1c32 adjusts the PR to create a new migration for the null Contributor work.

File: src/django/api/management/commands/delete_contributor.py (hunk 1 above)
Comment (kellyi) [reply src/django/api/management/commands/delete_contributor.py]: Sounds good. I renamed it in 
26f33df

File: src/django/api/management/commands/delete_contributor.py (hunk 2 above)
Comment (kellyi) [reply src/django/api/management/commands/delete_contributor.py]: Good point. I updated this in 
26f33df by adding separate nested try blocks for the Contributor and EmailAddress objects.

**Comment by kellyi** [comment]
Renamed the command in ea654bf261f137328c28b4e99f8ecb94b0a9cf0b
to be `delete_user_and_contributor` which more accurately describes what it's doing!

**Pull request merged**